src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs

```diff
@@ -140,23 +142,29 @@ private static void HandleBinaryOperator(in OperationAnalysisContext context, IN
             context.ReportDiagnostic(binary.Syntax.GetLocation().CreateDiagnostic(EqualityRule));
         }
 
-        private static void HandleInvocationOperation(in OperationAnalysisContext context, INamedTypeSymbol symbolType,
-            INamedTypeSymbol? symbolEqualityComparerType, ImmutableDictionary<string, ImmutableHashSet<INamedTypeSymbol>> equalityComparerMethods)
+        private static void HandleInvocationOperation(
+            in OperationAnalysisContext context,
+            INamedTypeSymbol symbolType,
+            INamedTypeSymbol? symbolEqualityComparerType,
+            ImmutableDictionary<string, ImmutableHashSet<INamedTypeSymbol>> equalityComparerMethods,
+            INamedTypeSymbol? systemHashCodeType)
         {
             var invocationOperation = (IInvocationOperation)context.Operation;
             var method = invocationOperation.TargetMethod;
 
             switch (method.Name)
             {
                 case WellKnownMemberNames.ObjectGetHashCode:
-                    if (IsNotInstanceInvocationOrNotOnSymbol(invocationOperation, symbolType))
+                    // This is a call for an instance of ISymbol.GetHashCode()
+                    // without the correct arguments
+                    if (IsSymbolType(invocationOperation.Instance, symbolType))
```

Comment:
IsSymbolType will crash if `invocationOperation.Instance` is null. This can happen if the static `object.GetHashCode` is called. Can you confirm and add a test?